### PR TITLE
Enable cancellation during file sync

### PR DESCRIPTION
**Pull request by shuo-wu**
longhorn/longhorn#2416

### CODE REVIEW

**Comment by shuo-wu** [comment]
Regression test: https://ci.longhorn.io/job/private/job/longhorn-tests-regression/155/

**Comment by innobead** [comment]
> Regression test: https://ci.longhorn.io/job/private/job/longhorn-tests-regression/155/

regression failed?

**Comment by shuo-wu** [comment]
> > Regression test: https://ci.longhorn.io/job/private/job/longhorn-tests-regression/155/
> 
> regression failed?

13 test cases are failed in the regression result. And right now there are around 12 test cases failed in night tests. I will check if some failures are caused by this PR later.

File: cli/ssync/main.go [hunk 1]
```diff
@@ -42,7 +43,8 @@ func Main() {
 		dstPath := args[0]
 
 		ops := &rest.SyncFileStub{}
-		err := rest.Server(*port, dstPath, ops)
+		ctx, cancelFunc := context.WithCancel(context.Background())
```
Comment (innobead) [inline cli/ssync/main.go:46]: It seems in this case, I assume there is no need to run cancelFunc to stop the server before the process finished. The change here is just to conform to the interface change.

**Review by innobead — APPROVED** [review]
LGTM

**Review by cclhsu — APPROVED** [review]

File: cli/ssync/main.go (hunk 1 above)
Comment (shuo-wu) [reply cli/ssync/main.go:46]: Yes. And this part is no longer used by any longhorn workloads. Now it's only for manually debugging the sync feature.

**Comment by shuo-wu** [comment]
Directly running the failed cases that the failure reasons are different from that of the nightly test gets passed. And the previous failure reasons seem to be not related to rebuilding/BIM. I think it's good to go.

**Pull request merged**